crates/starknet-server/tests/test_advancing_time.rs

```diff
@@ -31,31 +33,37 @@ mod advancing_time_tests {
             .send_custom_rpc("starknet_getBlockWithTxHashes", json!({ "block_id": "latest" }))
             .await["result"];
         assert!(set_time_block["timestamp"].as_u64() >= Some(past_time));
+        assert!(set_time_block["timestamp"].as_u64() < Some(past_time + BUFFER_TIME_SECONDS));
 
         // wait 1 second
         thread::sleep(time::Duration::from_secs(1));
 
-        // create block and check if block_timestamp is greater than past_time
+        // create block and check if block_timestamp is greater than past_time, check if it's inside
+        // buffer limit
         devnet.post_json("/create_block".into(), Body::from(json!({}).to_string())).await.unwrap();
         let empty_block = &devnet
             .send_custom_rpc("starknet_getBlockWithTxHashes", json!({ "block_id": "latest" }))
             .await["result"];
         assert!(empty_block["timestamp"].as_u64() > Some(past_time));
+        assert!(empty_block["timestamp"].as_u64() < Some(past_time + BUFFER_TIME_SECONDS));
 
         // wait 1 second
         thread::sleep(time::Duration::from_secs(1));
 
-        // check if after mint timestamp is greater than last block
+        // check if after mint timestamp is greater than last block, check if it's inside buffer
+        // limit
         devnet.mint(DUMMY_ADDRESS, DUMMY_AMOUNT).await;
         let mint_block = &devnet
             .send_custom_rpc("starknet_getBlockWithTxHashes", json!({ "block_id": "latest" }))
             .await["result"];
         assert!(mint_block["timestamp"].as_u64() > empty_block["block_timestamp"].as_u64());
+        assert!(mint_block["timestamp"].as_u64() < Some(past_time + BUFFER_TIME_SECONDS));
```

Comment:
Is `past_time` the correct variable here? Similar comment applicable to L107, L189, L204.

How about adding a util function that does this buffered ge comparison?
```rust
fn assert_ge_with_buffer(val1, val2) {
    assert!(val1 >= val2);
    assert!(val1 < val2 + BUFFER_TIME_SECONDS);
}
```

It's less code and less chance for an error.

Reply:
Added in 5f7f914dbd12d30511013922ee90a3bc54e503fc

Reply:
Resolving after this was added: https://github.com/0xSpaceShard/starknet-devnet-rs/commit/6b1947f0d05ae5273a38ac6d957b13c27800ba01

Reply:
and in 6b1947f0d05ae5273a38ac6d957b13c27800ba01